andela/ah-backend-prime
# #164798289 Email Verification

#### What does this PR do?
Ensures a user receives an activation email upon registration to verify the ownership of the email.

#### Description of Task to be completed?
- Write tests for email verification
- Add Authors haven's email details to settings file
- Add email message and recipients to registration route
- Add activation link in email

#### How should this be manually tested?

After cloning the repo, cd into it and run the following commands to activate the virtual environment
- `$ virtualenv venv`
- `$ source venv/bin/activate`

Install all requirements
- `$ pip install -r requirements.txt`

Make migrations
- `$ python manage.py makemigrations`
- `$ python manage.py migrate`

Run the server 
- `$ python manage.py runserver`

Open Postman and Register a user using the following format and providing a valid email address:
Add a request using this route

- `localhost:8000/api/users/register/`

Set the method to POST
Click on body, Select raw and choose JSON(application/json) as your data format
Pass your details in this format providing a valid gmail account
```
{
    "user": {
                  "email": "your email address",
                  "password": "your valid password",
                  "username": "your preferred username"
       }
}
```
Check your email and use the link provided to activate your account

After this you should be able to login to your account


#### What are the relevant pivotal tracker stories?
[#164798289](https://www.pivotaltracker.com/story/show/164798289)

#### Screenshots

- Postman

<img width="1123" alt="Screenshot 2019-04-03 at 10 47 35" src="https://user-images.githubusercontent.com/43245704/55461785-f3fc5d00-55fd-11e9-83f8-b0d21404c63a.png">

- Email
<img width="1436" alt="Screenshot 2019-04-04 at 22 29 53" src="https://user-images.githubusercontent.com/43245704/55583064-a2e98780-5729-11e9-934d-1c5d54950fb7.png">
